src/stdplus/misc.hpp

```diff
@@ -13,12 +13,40 @@ and stdplus/misc are assorted parts of that
 #include <memory>
 #include <string>
 #include <sstream>
+#include <limits>
 #include <boost/numeric/conversion/cast.hpp>
+#include <functional> // for less
 
+#include <limits>
+
+#include "tnetdbg.hpp" // debug
+#include "strings_utils_base.hpp" // to create to_debug()
+#include "utils/check.hpp"
 
 namespace stdplus {
 
 // ===========================================================================================================
+// exception types
+
+/// For exceptions that are not errors, but are expected
+/// alternative situation in program.
+class expected_exception : public std::exception {
+	public:
+		const char* what() const noexcept override;
+};
+
+/// Throw this if some element was not found, but this is normal situation.
+/// E.g. when public-key was not yet created at all (to differentiate
+/// from case where public-key was found but can not be loaded due to format errors etc)
+class expected_not_found : public stdplus::expected_exception {
+	public:
+		const char* what() const noexcept override;
+};
+
+template<typename T> constexpr bool templated_always_false() { return false; }
```

Comment:
documented why it was needed, and switched to the idea you recommended with ,FALSE template